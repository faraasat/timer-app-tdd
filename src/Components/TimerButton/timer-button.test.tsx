import { shallow } from "enzyme";
import PauseIcon from "@material-ui/icons/Pause";
import TimerButtonComponent from "./timer-button.component";

describe("Timer Button Shallow Render", () => {
  const mockParams = {
    onClick: jest.fn(),
    toggle: false,
    IconComponent: PauseIcon,
    color: "pink",
    id: "test-timer-button",
  };

  const shallowRender = shallow(<TimerButtonComponent {...mockParams} />);

  it("should match snapshot", () => {
    expect(shallowRender).toMatchSnapshot();
  });

  it("should render div(s)", () => {
    expect(shallowRender.find("div").length).toBeGreaterThanOrEqual(1);
  });

  it("should render a button", () => {
    expect(shallowRender.find("button").length).toEqual(1);
  });
});
