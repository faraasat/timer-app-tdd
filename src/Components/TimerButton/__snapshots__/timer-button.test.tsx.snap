// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer Button Shallow Render should match snapshot 1`] = `ShallowWrapper {}`;
